tools/fetch-metrics-for-space.js

```diff
@@ -0,0 +1,102 @@
+import {
+  DynamoDBClient,
+  QueryCommand
+} from '@aws-sdk/client-dynamodb'
+import { unmarshall } from '@aws-sdk/util-dynamodb'
+
+export async function fetchMetricsForSpaceCmd () {
+  const {
+    ENV,
+    SPACE_DID,
+    TABLE_NAME,
+  } = getEnv()
+
+  const { client, tableName } = getDynamoDb(
+    TABLE_NAME,
+    ENV,
+    getRegion(ENV)
+  )
+
+  const rows = await getAllTableRows(client, tableName, SPACE_DID)
+  console.log(`Metrics found for provided space DID: ${rows.length}`)
+  for (const row of rows) {
+    console.log(`${row.name}: ${row.value}`)
+  }
+}
+
+/**
+ * @param {import('@aws-sdk/client-dynamodb').DynamoDBClient} dynamo
+ * @param {string} tableName
+ * @param {string} space
+ * @param {object} [options]
+ * @param {number} [options.limit]
+ */
+export async function getAllTableRows (dynamo, tableName, space, options = {}) {
+  const cmd = new QueryCommand({
+    TableName: tableName,
+    Limit: options.limit || 30,
+    KeyConditions: {
+      space: {
+        ComparisonOperator: 'EQ',
+        AttributeValueList: [{ S: space }],
+      },
+    }
+  })
+  
+  const response = await dynamo.send(cmd)
+  return response.Items?.map(i => unmarshall(i)) || []
+}
+
+/**
+ * Get Env validating it is set.
+ */
+function getEnv() {
+  return {
+    ENV: mustGetEnv('ENV'),
+    SPACE_DID: mustGetEnv('SPACE_DID'),
+    TABLE_NAME: mustGetEnv('TABLE_NAME'),
+  }
+}
+
+/**
+ * 
+ * @param {string} name 
+ * @returns {string}
+ */
+function mustGetEnv (name) {
+  if (!process.env[name]) {
+    throw new Error(`Missing env var: ${name}`)
+  }
+
+  // @ts-expect-error there will always be a string there, but typescript does not believe
+  return process.env[name]
+}
+
+/**
+ * @param {string} env
+ */
+function getRegion (env) {
+  if (env === 'staging') {
+    return 'us-east-2'
+  }
+
+  return 'us-west-2'
+}
+
+/**
+ * @param {string} tableName
+ * @param {string} env
+ * @param {string} region
+ */
+function getDynamoDb (tableName, env, region) {
```

Comment:
How does it authenticate?

Reply:
Do you mean authenticate to run the action? This is only for admin usage at the moment. We will integrate this with API later on (with `space/info`, or other API) where users will be able to interact with the system. Github triggers can only be run by collaborators with run permissions in repo.

Or do you mean dynamoDB? Yeah, I guess we need to put AWS credentials in... In my env I have them configured, and seed.run (where integration tests run with similar setup) also should have them set. I will add them

Reply:
yes the latter!